zipline/assets/asset_writer.py

```diff
@@ -563,3 +560,9 @@ def _load_data(self, equities, futures, exchanges, root_symbols):
             exchanges=exchanges_output,
             root_symbols=root_symbols_output,
         )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        pass  # Shouldn't dispose of the engine explicitly
```

Comment:
Well I took another look at the overall docs for the sqlalchemy engine, and it said that the engine shouldn't be disposed until the end of the program because this prevents any future queries to the database. 
As of right now, the readers don't try to explicitly create an engine, so closing the engine would cause the readers to fail.

 I agree that the change wouldn't be necessary, as of right now